### PR TITLE
Add an ordering step to the oefdb_format command

### PR DESCRIPTION
**Pull request by GeeWee**
[Notion card](https://www.notion.so/climatiq/Data-workflow-Automatically-standardize-the-format-of-OEFDB-csv-a6ff6f26fcd6413f9618769b185bf0a8)

This PR adds an ordering step to the `oefdb_format` command.

It is "almost" consistent with the formatting from LibreOffice but not entirely. I've spent several hours troubleshooting why it is, and I cannot seem to make it behave the same way. I have checked out sort stability, but in the end it's not very transparent how LibreOffice and others _actually_ sort. U

However the solution I've gotten behaves _almost_ like LIbreOffice and Excel, and the team has agreed that that's good enough for now (see the card)

# Other changes
- This PR also takes an excerpt out of the oefdb as saved by libreoffice and uses it as a test fixture for formatting and ordering. A snapshot has been saved for comparison, by manually running the file through `oefdb_format`. It seems that currently libreoffice saves the oefdb in the exact way our tooling expects, so in this case the formatting is a no-op.
- It also takes a snapshot of a csv saved by gsheets and performs the same.
- Updated the poetry.lock file

### CODE REVIEW

File: tests/test_formatting_and_ordering.py [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+        # check formatted file is equal to the previously generated snapshot
+        are_files_equal = filecmp.cmp(
+            "fixtures/snapshots/valid_oefdb_excerpt.csv",
+            csv_file.name,
+            shallow=False,
+        )
+        assert are_files_equal
+
+
+def test_format_and_sort_oefdb_excerpt_gsheets():
```
Comment (GeeWee) [inline tests/test_formatting_and_ordering.py:86]: and gsheets

**Comment by GeeWee** [comment]
@motin here's a PR that looks at ordering of the OEFDB, pls review

**Review by motin — APPROVED** [review]
Looks mergable :) Left some optional cleanup comments

File: pytest.ini [hunk 2]
```diff
@@ -1,3 +1,3 @@
 [pytest]
-addopts = --cov=oefdb --cov-report term-missing --verbose
+;addopts = --cov=oefdb --cov-report term-missing --verbose
```
[the remark this answers is not in the document]
Comment (GeeWee) [reply pytest.ini]: Ah, it breaks pytest-watch I think. I normally just turn it off temporarily and forgot to uncomment it

File: tests/test_formatting_and_ordering.py [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+
+        assert result.exit_code == 0
+
+        with open(csv_file.name) as f:
+            updated_contents = f.read()
+            # Contents have been updated with:
+            # 1. Newlines changed to \n from \r\n
+            # 2. Quoted strings have been unquoted
+            # 3. A trailing newline has been added
+            # 4. The rows have switched places (due to reordering)
```
[the remark this answers is not in the document]
Comment (GeeWee) [reply tests/test_formatting_and_ordering.py:45]: Yes good catch. I've changed the input ordering

**Pull request merged**